### PR TITLE
Added level tooltips and one safari level achievement

### PR DESCRIPTION
**Pull request by RegisCoaxans**
## Description
- New tooltips for quest and safari : they tell the players about the XP amounts.
- Safari level requirement so we can lock things behind defined levels.
- Safari level achievement to introduce the requirement. More to come later, if needed.
- safari level display : the tooltip indicates that the max level has been reached, the progress bar stays filled whenever such thing happens.

## Motivation and Context
- Safari and quest levels look less obscur.
- Eugene needed a safari level requirement.

## How Has This Been Tested?
- Changed my levels to see how it acts.

## Types of changes
- New feature
- UI improvement

### CODE REVIEW

File: src/scripts/achievements/SafariLevelRequirement.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    constructor(levelRequired: number) {
+        super(levelRequired, GameConstants.AchievementOption.more, GameConstants.AchievementType.Safari);
+    }
+
+    public getProgress() {
+        return Math.min(Safari.safariLevel(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `Needs safari level ${this.requiredValue}.`;
```
Comment (Farboo171) [inline src/scripts/achievements/SafariLevelRequirement.ts:11]:
```suggestion
        return `Needs Safari Level ${this.requiredValue}.`;
```

**Review by Farboo171 — CHANGES_REQUESTED** [review]
Minor capitalization change.
Tested the tooltips and achievement, looks good!

File: src/scripts/safari/Safari.ts [hunk 2]
```diff
@@ -487,8 +492,19 @@ class Safari {
...
+            tooltip.title = 'Max level reached';
+        } else {
+            tooltip.title = `${Safari.safariExp() - Safari.expRequiredForLevel(level)} / ${Safari.expRequiredForLevel(level + 1) - Safari.expRequiredForLevel(level)}`;
+        }
+        return tooltip;
+    }
+
     private static expRequiredForLevel(level: number) {
-        return Math.ceil(2000 * (1.2 ** (level - 1) - 1));
+        return Math.ceil(2000 * (1.2 ** (Math.min(Safari.maxSafariLevel, level) - 1) - 1));
```
Comment (Aegyo) [inline src/scripts/safari/Safari.ts:507]: Is this change needed?

Comment (RegisCoaxans) [reply src/scripts/safari/Safari.ts]: No. I tried to fulfill the xp bar that way before changing my mind.
Back to normal.

**Review by Aegyo — APPROVED** [review]
